### PR TITLE
Added implementation for valid parentheses in string

### PR DESCRIPTION
**Pull request by ajinkyac03**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

Fixes https://github.com/TheAlgorithms/Python/issues/6354

### CODE REVIEW

**Review by nuhmanpk — APPROVED** [review]

**Comment by ajinkyac03** [comment]
@cclauss please have a look.

**Comment by cclauss** [comment]
Duplicate of https://github.com/TheAlgorithms/Python/blob/master/data_structures/stacks/balanced_parentheses.py ?

File: strings/valid_parentheses.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                if not stack:
+                    return False
+                elif stack.pop() != valid_pairs[i]:
+                    return False
+                else:
+                    continue
+        if len(stack) == 0:
+            return True
+        else:
+            return False
```
Comment (cclauss) [inline strings/valid_parentheses.py:29]:
```suggestion
        return not stack
```

**Review by Anash3 — APPROVED** [review]

**Comment by cclauss** [comment]
How is this algorithm better than our existing [balanced_parentheses.py](https://github.com/TheAlgorithms/Python/blob/master/data_structures/stacks/balanced_parentheses.py)?

**Comment by tianyizheng02** [comment]
@cclauss Close this PR? It's stale, and it's a duplicate implementation like you said

**Pull request closed without merging**